src/v/kafka/server/handlers/produce.cc

```diff
@@ -207,6 +208,19 @@ static partition_produce_stages partition_append(
     };
 }
 
+ss::future<produce_response::partition> finalize_request_with_error_code(
+  error_code ec, std::unique_ptr<ss::promise<>>&& dispatch, model::ntp ntp) {
+    // submit back to promise source shard
+    (void)ss::smp::submit_to(
+      ss::this_shard_id(), [dispatch = std::move(dispatch)]() mutable {
```

Comment:
this is incorrect, should accept shard as an argument

Reply:
thanks!

Reply:
nice catch

Reply:
Consider `ssx::background` instead of `(void)...`.

Reply:
done